docs/azure_workload_identity.md

```diff
@@ -0,0 +1,167 @@
+
+Azure User-Assigned Managed Identities are configured with the Issuer URL (public URL of the Azure Blob Container) containing the public key and OpenID configuration to allow for federation of the identity to the OIDC identity provider. Every in-cluster component receives a Kubernetes Secret containing the details of a User-Assigned Managed Identity created on the component's behalf. The Azure Identity SDK used within those components requests an Azure AD access token for the identity specified by the configuration secret and provides the signed ServiceAccount token to Azure AD. Azure AD validates that the ServiceAccount token is from a trusted identity provider before issuing an access token for the User-Assigned Managed Identity.
+
+### Changes in the Credentials Secret with Azure AD Workload Identity
+
+Without Azure AD Workload Identity, an Azure credentials secret includes an `azure_client_secret` which is essentially a password used to authenticate with Azure AD for the identity identified by `azure_client_id`. This `azure_client_secret` needs to be kept secure and is not rotated automatically.
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_client_secret: <client secret>
+  azure_region: <region>
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+With Azure AD Workload Identity, an Azure credentials secret contains the `azure_client_id` of the User-Assigned Managed Identity that the component will be authenticating as along with the path to the projected ServiceAccount token, `azure_federated_token_file`. The ServiceAccount token mounted at this path is refreshed hourly and thus the credentials are short-lived.
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_federated_token_file: </path/to/mounted/service/account/token>
+  azure_region: <region>
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+### Steps to install an OpenShift Cluster with Azure AD Workload Identity
+
+1. Set the `$RELEASE_IMAGE` environment variable.
+
+   `$RELEASE_IMAGE` should be a recent and supported  OpenShift release image that you want to deploy in your cluster.
+   Please refer to the [support matrix](../README.md#support-matrix) for compatibilities.
+
+   A sample release image would be `RELEASE_IMAGE=quay.io/openshift-release-dev/ocp-release:${RHOCP_version}-${Arch}`
+
+   Where `RHOCP_version` is the OpenShift version (e.g `4.14.0-fc.4` or `4.14.3`) and the `Arch` is the architecture type (e.g `x86_64`).
+
+2. Extract Azure CredentialsRequests objects from the release image.
+
+   ```
+   oc adm release extract --cloud=azure --credentials-requests $RELEASE_IMAGE --to=./credreqs
+   ```
+
+3. Extract the `openshift-install` and `ccoctl` binaries from the release image.
+
+   ```
+   oc adm release extract --command=openshift-install $RELEASE_IMAGE
```

Comment:
Here we need to confirm that if the pull-secret is not specified, can not to extract openshift-install:
`[hmx@fedora auth]$ oc adm release extract --command=openshift-install registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-07-11-092038`
```
Warning: the default reading order of registry auth file will be changed from "${HOME}/.docker/config.json" to podman registry config locations in the future version of oc. "${HOME}/.docker/config.json" is deprecated, but can still be used for storing credentials as a fallback. See https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md for the order of podman registry config locations.
Warning: the default reading order of registry auth file will be changed from "${HOME}/.docker/config.json" to podman registry config locations in the future version of oc. "${HOME}/.docker/config.json" is deprecated, but can still be used for storing credentials as a fallback. See https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md for the order of podman registry config locations.
error: unable to read image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:7c76f09717ef38a6692b100784ae52c3f6e87a241a3635754f9189957640d79a: unauthorized: access to the requested resource is not authorized
```

I usually add pull-secret for authentication.
```
[hmx@fedora auth]$ oc adm release extract --command=openshift-install registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-07-11-092038 -a ~/.pull-secret

[hmx@fedora auth]$ ll | grep openshift-install
-rwxr-xr-x. 1 hmx hmx 632181720  7月11日 13:32 openshift-install

```

Also, is it necessary to extract oc.

Reply:
[cf20a62](https://github.com/openshift/cloud-credential-operator/pull/566/commits/cf20a62af76fa90b18b38ed32aef4d9fd0f14d92) adds a step for obtaining `oc`.